src/components/Clock.js

```diff
@@ -2,9 +2,8 @@ import React from 'react';
 
 class Clock extends React.Component {
   render() {
-    // const {date} = this.props;   
     return (
-      <p>Právě je {this.state.date.toLocalTimeString()}</p> // TypeError: Cannot read property 'date' of null
+      <p>Právě je {this.props.date && this.props.date.toLocaleTimeString()}</p>
```

Comment:
jo tady si zavolas ten state z App.js co jsi predala pres props. Ale zavolas si to z toho s `toLocaleTimeString()` Znamena to tedy ze cas se uzivateli zobrazi v jeho preferovanem lokalnim nastaveni. to muzes nastavit v tom new Date. vice zde https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString

Reply:
jo, `this.props.date &&` to je tam pro to, aby to nehazelo chybu, ze to je null. tohle vysvetlim na lekci :-)